Mathlib/Analysis/SpecialFunctions/Integrals.lean

```diff
@@ -453,25 +484,49 @@ theorem integral_exp_mul_complex {c : ℂ} (hc : c ≠ 0) :
   · fun_prop
 
 @[simp]
-theorem integral_log (h : (0 : ℝ) ∉ [[a, b]]) :
-    ∫ x in a..b, log x = b * log b - a * log a - b + a := by
-  have h' := fun x (hx : x ∈ [[a, b]]) => ne_of_mem_of_not_mem hx h
-  have heq := fun x hx => mul_inv_cancel₀ (h' x hx)
-  convert integral_mul_deriv_eq_deriv_mul (fun x hx => hasDerivAt_log (h' x hx))
-    (fun x _ => hasDerivAt_id x) (continuousOn_inv₀.mono <|
-      subset_compl_singleton_iff.mpr h).intervalIntegrable
-        continuousOn_const.intervalIntegrable using 1 <;>
-  simp [integral_congr heq, mul_comm, ← sub_add]
-
-@[simp]
-theorem integral_log_of_pos (ha : 0 < a) (hb : 0 < b) :
-    ∫ x in a..b, log x = b * log b - a * log a - b + a :=
-  integral_log <| not_mem_uIcc_of_lt ha hb
-
-@[simp]
-theorem integral_log_of_neg (ha : a < 0) (hb : b < 0) :
-    ∫ x in a..b, log x = b * log b - a * log a - b + a :=
-  integral_log <| not_mem_uIcc_of_gt ha hb
+theorem integral_log : ∫ s in a..b, log s = b * log b - a * log a - b + a := by
+  -- Simple case one: x is positive, y is zero
+  have case₁ {t : ℝ} (ht : 0 < t) : ∫ s in (0)..t, log s = t * log t - t := by
+    -- Compute the integral by giving a primitive and considering it limit as x
+    -- approaches 0 from the right. The following lines were suggested by Gareth
+    -- Ma on Zulip.
+    rw [integral_eq_sub_of_hasDerivAt_of_tendsto (f := fun x ↦ x * log x - x)
+      (fa := 0) (fb := t * log t - t)]
+    · abel
+    · exact ht
+    · intro s hs
+      norm_num at hs
+      convert (hasDerivAt_mul_log hs.left.ne.symm).sub (hasDerivAt_id s) using 1
+      abel
+    · exact intervalIntegrable_log'
+    · have := tendsto_log_mul_rpow_nhds_zero zero_lt_one
+      simp_rw [rpow_one, mul_comm] at this
+      convert this.sub (tendsto_nhdsWithin_of_tendsto_nhds Filter.tendsto_id)
+      abel
+    · apply tendsto_nhdsWithin_of_tendsto_nhds
+      apply ContinuousAt.tendsto
+      fun_prop
+
+  -- Simple case two: y is zero
+  have case₂ {t : ℝ} : ∫ s in (0)..t, log s = t * log t - t := by
+    rcases lt_trichotomy t 0 with h|h|h
+    · -- If t is negative, use that log is an even function to reduce to the positive case.
+      conv => arg 1; arg 1; intro t; rw [← log_neg_eq_log]
+      rw [intervalIntegral.integral_comp_neg]
```

Comment:
You can combine subsequent calls to `rw`, like `rw [foo, bar]`. Also below.

Reply:
Implemented. Thanks!